spring-grpc-spring-boot-autoconfigure/src/test/java/org/springframework/grpc/autoconfigure/common/GrpcCodecConfigurationTest.java

```diff
@@ -0,0 +1,82 @@
+package org.springframework.grpc.autoconfigure.common;
+
+import io.grpc.Compressor;
+import io.grpc.CompressorRegistry;
+import io.grpc.Decompressor;
+import io.grpc.DecompressorRegistry;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.grpc.autoconfigure.common.codec.GrpcCodecConfiguration;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link GrpcCodecConfiguration}.
+ *
+ * @author Andrei Lisa
+ */
+public class GrpcCodecConfigurationTest {
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+		.withConfiguration(AutoConfigurations.of(GrpcCodecConfiguration.class));
+
+	@Test
+	void testCompressorRegistryBean() {
+		contextRunner.run(context -> {
+			assertThat(context).hasSingleBean(CompressorRegistry.class);
+			CompressorRegistry registry = context.getBean(CompressorRegistry.class);
+			assertThat(registry).isNotNull();
+		});
+	}
+
+	@Test
+	void testDecompressorRegistryBean() {
+		contextRunner.run(context -> {
+			assertThat(context).hasSingleBean(DecompressorRegistry.class);
+			DecompressorRegistry registry = context.getBean(DecompressorRegistry.class);
+			assertThat(registry).isNotNull();
+		});
+	}
+
+	@TestConfiguration
```

Comment:
So this configuration is not needed. The `@TestConfiguration` is only used during autoconfiguration scanning when using `@SpringBootTest` (details [here](https://docs.spring.io/spring-boot/reference/testing/spring-boot-applications.html#testing.spring-boot-applications.detecting-configuration)).

If you remove this class you will see the test passes as the registries come from your newly introduced code AC.

Reply:
done, Thanks for the details about the `@TestConfiguration`